app/tests/integration/nimbus/test_mobile_integration.py

```diff
@@ -0,0 +1,158 @@
+    return locale_list
+
+
+def client_info_list():
+    def poll_job(s, redash_url, job):
+        while job["status"] not in (3, 4):
+            response = s.get("{}/api/jobs/{}".format(redash_url, job["id"]))
+            job = response.json()["job"]
+            time.sleep(1)
+
+        if job["status"] == 3:
+            return job["query_result_id"]
+
+        return None
+
+    redash_url = "https://sql.telemetry.mozilla.org"
+    query_id = 86023
+    s = requests.Session()
+    s.headers.update(
+        {"Authorization": "Key {}".format("TItInJieogEryYnzAFOeA5RMBBzYDN3nc59mqren")}
+    )
+    payload = dict(max_age=0, parameters={"something_cool": 1})
+    response = s.post(
+        "{}/api/queries/{}/results".format(redash_url, query_id), data=json.dumps(payload)
+    )
+    result_id = poll_job(s, redash_url, response.json()["job"])
+    response = s.get(
+        "{}/api/queries/{}/results/{}.json".format(redash_url, query_id, result_id)
+    )
+    return response.json()["query_result"]["data"]["rows"]
+
+
+@pytest.fixture
+def load_app_context():
+    def _load_app_context_helper(context):
+        base_app_context = BaseAppContextDataClass.from_dict(context)
+        return nimbus.AppContext(
+            app_id=base_app_context.app_id,
+            app_name=base_app_context.app_name,
+            channel=base_app_context.channel,
+            app_version=base_app_context.app_version,
+            app_build=base_app_context.app_build,
+            architecture=base_app_context.architecture,
+            device_manufacturer=base_app_context.device_manufacturer,
+            device_model=base_app_context.device_model,
+            locale=base_app_context.locale,
+            os=base_app_context.os,
+            os_version=base_app_context.os_version,
+            android_sdk_version=base_app_context.android_sdk_version,
+            debug_tag=base_app_context.debug_tag,
+            installation_date=base_app_context.installation_date,
+            home_directory=base_app_context.home_directory,
+            custom_targeting_attributes=base_app_context.custom_targeting_attributes,
+        )
+
+    return _load_app_context_helper
+
+
+@pytest.fixture(name="ar_units")
+def fixture_ar_units():
+    return nimbus.AvailableRandomizationUnits(None, 0)
+
+
+@pytest.fixture(name="sdk_client")
+def fixture_sdk_client(ar_units):
+    def _client_helper(app_context):
+        return nimbus.NimbusClient(app_context, os.getcwd(), None, ar_units)
+
+    return _client_helper
+
+
+@pytest.fixture(name="jexl_evaluator")
+def fixture_jexl_evaluator():
+    def _eval_jexl(sdk_client, expression):
+        targetting_helper = sdk_client.create_targeting_helper()
+        try:
+            value = targetting_helper.eval_jexl(expression)
+        except nimbus.NimbusError.EvaluationError:
+            return None
+        else:
+            return value
+
+    return _eval_jexl
+
+
+@pytest.mark.parametrize("context", client_info_list())

```

Comment:
Yeah testing the JEXL expression from the mobile advanced targetings is specifically what I'd like to test here since Experimenter has no way of determining whether they're 'valid' or not, only the SDK can determine that.